engine/access/relay/engine_test.go

```diff
@@ -0,0 +1,81 @@
+package relay
+
+import (
+	"testing"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/onflow/flow-go/module"
+	mockmodule "github.com/onflow/flow-go/module/mock"
+	"github.com/onflow/flow-go/network"
+	"github.com/onflow/flow-go/network/mocknetwork"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type Suite struct {
+	suite.Suite
+
+	engine   *Engine
+	me       module.Local
+	channels network.ChannelList
+	conduits map[network.Channel]*mocknetwork.Conduit
+}
+
+func (suite *Suite) SetupTest() {
+	suite.channels = network.ChannelList{
+		network.Channel("test-channel-1"),
+	}
+	net := new(mockmodule.Network)
+	unstakedNet := new(mockmodule.Network)
+	suite.me = new(mockmodule.Local)
+	suite.conduits = make(map[network.Channel]*mocknetwork.Conduit)
+
+	for _, channel := range suite.channels {
+		con := new(mocknetwork.Conduit)
+		suite.conduits[channel] = con
+		net.On("Register", channel, mock.Anything).Return(new(mocknetwork.Conduit), nil).Once()
+		unstakedNet.On("Register", channel, mock.Anything).Return(con, nil).Once()
+	}
+
+	eng, err := New(
+		zerolog.Logger{},
+		suite.channels,
+		net,
+		unstakedNet,
+		suite.me,
+	)
+	suite.Require().Nil(err)
+
+	suite.engine = eng
+}
+
+func TestRelayEngine(t *testing.T) {
+	suite.Run(t, new(Suite))
+}
+
+func getEvent() interface{} {
+	return struct {
+		foo string
+	}{
+		foo: "bar",
+	}
+}
+
+// TestHappyPath tests that the relay engine relays events for each
+// channel that it was created with
+func (suite *Suite) TestHappyPath() {
+	for channel, conduit := range suite.conduits {
+		id := unittest.IdentifierFixture()
+		event := getEvent()
+
+		conduit.On("Publish", event, mock.AnythingOfType("flow.Identifier")).Return(nil).Once()
```

Comment:
shouldn't we expect the target to be `flow.ZeroID`
```suggestion
		conduit.On("Publish", event, flow.ZeroID).Return(nil).Once()
```